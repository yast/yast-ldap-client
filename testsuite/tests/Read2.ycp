/**
 *  Read.ycp
 *  Test of Ldap:Read function
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{
    // testedfiles: Ldap.ycp

    import "Ldap";
    import "Testsuite";

    map READ = $[
	"etc": $[
	    "nsswitch_conf" : $[
		"passwd"	: "sss",
		"group"		: "sss",
		"passwd_compat"	: nil,
		"group_compat"	: nil
	    ],
	    "ldap_conf" : $[
		"v" : $[
		    "/etc/ldap.conf": $[
			"host" : "localhost",
			"base" : "dc=suse,dc=cz",
			"nss_base_passwd" : nil,
			"nss_base_shadow" : nil,
			"nss_base_group" : nil,
			"ldap_version": nil,
			"ssl": nil,
			"pam_password": "crypt",
			"tls_cacertdir"	: "/etc/openldap/cacerts/",
			"tls_cacertfile": nil,
			"tls_checkpeer" : "no",
			"uri"	: "ldap://localhost:333"
		    ]
		]
	    ],
	    "krb5_conf" : $[
		"v" : $[
		    "libdefaults": $[
			"default_realm": ["SUSE.CZ"],
		    ],
		    "SUSE.CZ": $[
			"kdc": ["kdc.suse.cz"],
		    ],
		],
	    ],
	    // /etc/security/pam_*
	    "security": $[
		"section": $[
		    "/etc/security/pam_unix2.conf": $[],
		],
		"v": $[
		    "/etc/security/pam_unix2.conf": $[
			"auth"	: ""
		    ],
		]
	    ],
	    "sssd_conf"	: $[
		"v"			: $[
		    "domain/default"	: $[
			"krb5_realm"	: "SUSE.CZ",
			"krb5_kdcip"	: "kdc.suse.cz",
			"ldap_schema"	: nil,
			"cache_credentials"	: "True",
			"enumerate"	: nil,
                        "ldap_id_use_start_tls" : nil,
                        "ldap_user_search_base" : nil,
                        "ldap_group_search_base"        : "ou=group,dc=suse,dc=cz"
		    ],
		],
	    ],
	],
	"sysconfig": $[
	    "ldap": $[
		"BASE_CONFIG_DN": nil,
		"BIND_DN": "uid=manager,dc=suse,dc=cz",
		"FILE_SERVER": "no"
	    ]
	],
	"init": $[
	    "scripts": $[
		"exists": false,
	    ],
	],
	"passwd": $[
	    "passwd": $[
		"plusline": "+",
		"pluslines": ["+"]
	    ]
	],
	"product" : $[ "features" :
	    $[ "EVMS_CONFIG" : "nazdar" ],
	],
	"target"	: $[
	    "size"	: -1,
	    "stat"	: $[ 1:2 ],
	],
    ];

    map EX = $[
	"target": $[
	    "bash": 0,
	    "bash_output": $[
		"stdout"	: "",
	    ]
	],
	"passwd": $[
	    "init": true
	]
    ];

    Testsuite::Init ([READ,$[],$[]], nil);

    Testsuite::Test (``(Ldap::Read ()), [ READ, $[], EX ], 0);

    Testsuite::Dump ( sformat ("ldap: -%1-", Ldap::start) );

    Testsuite::Dump ( sformat ("sssd: -%1-", Ldap::sssd) );

    Testsuite::Dump ( sformat ("nss: -%1-", Ldap::nss_base_passwd) );
    Testsuite::Dump ( sformat ("nss: -%1-", Ldap::nss_base_group) );

}
