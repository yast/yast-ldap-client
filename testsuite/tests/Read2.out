Read	.etc.ldap_conf.v."/etc/ldap.conf"."uri" "ldap://localhost:333"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."base" "dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."ssl" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_cacertdir" "/etc/openldap/cacerts/"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_cacertfile" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."tls_checkpeer" "no"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_passwd" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_shadow" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_base_group" nil
Read	.etc.ldap_conf.v."/etc/ldap.conf"."pam_password" "crypt"
Read	.sysconfig.ldap.BASE_CONFIG_DN nil
Read	.sysconfig.ldap.FILE_SERVER "no"
Read	.sysconfig.ldap.BIND_DN "uid=manager,dc=suse,dc=cz"
Read	.etc.ldap_conf.v."/etc/ldap.conf"."nss_map_attribute" 0
Execute	.passwd.init $["base_directory":"/etc"] true
Read	.passwd.passwd.pluslines ["+"]
Read	.etc.sssd_conf.v."domain/default"."krb5_realm" "SUSE.CZ"
Read	.etc.sssd_conf.v."domain/default"."krb5_kdcip" "kdc.suse.cz"
Read	.etc.sssd_conf.v."domain/default"."ldap_schema" nil
Read	.etc.sssd_conf.v."domain/default"."cache_credentials" "True"
Read	.etc.sssd_conf.v."domain/default"."enumerate" nil
Read	.etc.sssd_conf.v."domain/default"."ldap_id_use_start_tls" nil
Read	.etc.sssd_conf.v."domain/default"."ldap_user_search_base" nil
Read	.etc.sssd_conf.v."domain/default"."ldap_group_search_base" "ou=group,dc=suse,dc=cz"
Return	true
Dump	ldap: -true-
Dump	sssd: -true-
Dump	nss: -dc=suse,dc=cz-
Dump	nss: -ou=group,dc=suse,dc=cz-
